### PR TITLE
[SSO] refactor request_new_domain method

### PR DESCRIPTION
**Pull request by biyeun**
## Summary
It would be nice if `request_new_domain` did not require that some random form be passed into it. I refactored the signature to explicitly require a `project_name` and made sure to update the two usages to pass in the correct form field. I also noticed that `domain_timezone` was not a field in any of the forms for the two usages (nor is it a field in ANY form), so I removed it.

Tested locally for both the sign-up user workflow and create new domain workflow. All fine.

I am doing this because I need `request_new_domain` to not require a form for the SSO Create User work I'm doing. I made this into a separate PR since it's a small change in a high risk area and i didn't want it being obfuscated by any SSO specific changes

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
None

### QA Plan
Not needed. very small isolated change and tested where I'm very confident nothing breaks.

### Safety story
Small, isolated change. tested the two workflows that use this method locally and there were no adverse effects.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/registration/utils.py [hunk 1]
```diff
@@ -108,9 +107,6 @@ def request_new_domain(request, form, is_new_user=True):
         # Avoid projects created by dimagi.com staff members as self started
         new_domain.internal.self_started = not current_user.is_dimagi
 
-        if form.cleaned_data.get('domain_timezone'):
-            new_domain.default_timezone = form.cleaned_data['domain_timezone']
```
Comment (millerdev) [inline corehq/apps/registration/utils.py:112]: Why was this removed?

**Review by millerdev — COMMENTED** [review]
Nice refactor. I'd like to approve, but the timezone question seems like reason to block unless it's essentially dead code (e.g., `default_timezone` is never present).

File: corehq/apps/registration/utils.py (hunk 1 above)
Comment (biyeun) [reply corehq/apps/registration/utils.py:112]: I mention the reason both in the commit message and the PR description.
>  I also noticed that domain_timezone was not a field in any of the forms for the two usages (nor is it a field in ANY form), so I removed it.

**Review by joxl — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

File: corehq/apps/registration/utils.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/registration/utils.py:112]: Sorry, missed that.
